### PR TITLE
Implement keepalive plugin

### PR DESCRIPTION
**Pull request by Hzj-jie**
This  change implements a KeepAlive plugin to keep system alive. The FR was discussed in #2204.

I have tested on Kindle. It looks good. I will try it on Kobo soon.

### CODE REVIEW

File: plugins/keepalive.koplugin/main.lua [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local _ = require("gettext")
+
+local KeepAlive = WidgetContainer:new{
+    name = "keepalive",
+}
+
+function KeepAlive:init()
+    if Device:isKobo() then
+        self.enable = function() UIManager:_stopAutoSuspend() end
```
Comment (houqp) [inline plugins/keepalive.koplugin/main.lua]: self.enable and self.disable can be initialized as a local function at the module level so you don't have to create two new anonymous functions every time `KeepAlive:init` is called. In fact, you can also initialize the menu callback based on the current device model only once.

**Comment by Hzj-jie** [comment]
The implementation works well on Kobo.

File: plugins/keepalive.koplugin/main.lua [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+local function showConfirmBox(disable)
+    UIManager:show(ConfirmBox:new{
+        text = _("The system won't sleep when this message is showing.\nPress \"Stay Alive\" if you prefer to keep system on even after closing this notification. *It will drain the battery.*\n\nIf for any reasons KOReader died before \"Close\" is pressed, please start and close KeepAlive plugin again to ensure settings are reset."),
+        ok_text = _("Close"),
+        ok_callback = disable,
+        cancel_text = _("Stay Alive"),
+    })
+end
+
+local keepAliveText = _("Keep Alive")
```
Comment (houqp) [inline plugins/keepalive.koplugin/main.lua]: This one is not used?

Comment (Hzj-jie) [inline plugins/keepalive.koplugin/main.lua]: No. Updated.

**Comment by retrue** [comment]
Two little remarks.
One, the menu entry, **Keep Alive**, should be **Keep alive** for stylistic coherence.
Two, when the "Keep alive" plugin is active it should show the menu entry ticked, to know at a glance if it is active or not.

**Comment by houqp** [comment]
Agree with the `Keep alive` naming suggestion. The menu entry tick is not needed because the current implementation will stop the keep alive mode when user close the pop up as explained in the pop up message :)

**Comment by retrue** [comment]
@houqp I have just checked again. The current implementation will maintain the keep alive mode when the user close the pop up after pushing the button "Stay Alive". A menu entry ticked would remind the user that the plugin is active.

**Comment by houqp** [comment]
My bad, I missed the stay alive button. Yeah, you are right, it's better to have a checkmark for the menu entry :)

**Comment by houqp** [comment]
Feel free to open a ticket for this.

**Pull request merged**